internal/pkg/composable/providers/docker/docker.go

```diff
@@ -131,20 +131,22 @@ func generateData(event bus.Event) (*dockerContainerData, error) {
 		container: container,
 		mapping: map[string]interface{}{
 			"container": map[string]interface{}{
-				"id":     container.ID,
-				"name":   container.Name,
-				"image":  container.Image,
+				"id":   container.ID,
+				"name": container.Name,
+				"image": map[string]interface{}{
+					"name": container.Image,
+				},
 				"labels": labelMap,
 			},
 		},
 		processors: []map[string]interface{}{
 			{
 				"add_fields": map[string]interface{}{
 					"fields": map[string]interface{}{
-						"id":     container.ID,
-						"name":   container.Name,
-						"image":  container.Image,
-						"labels": processorLabelMap,
+						"id":         container.ID,
+						"name":       container.Name,
+						"image.name": container.Image,
```

Comment:
Will this end up to be a nested object in ES? I have some concerns that it might not so I would propose to either test or just use a nested map here like what you do in the `mapping`.

Reply:
`add_fields` processor support this type of field value, see the [example](https://github.com/elastic/beats/blob/main/libbeat/processors/actions/add_fields_test.go#L104-L114)

yes, it is added as an nested object in es:
<img width="1739" alt="Screenshot 2023-03-29 at 09 51 09" src="https://user-images.githubusercontent.com/28299531/228464346-4343d74c-c465-443e-9605-79e9c6912190.png">